test/middleware/spec.back-link.js

```diff
@@ -1,11 +1,18 @@
-var backLinks = require('../../lib/middleware/back-links');
-
-var util = require('util'),
+var proxyquire = require('proxyquire'),
+    util = require('util'),
     EventEmitter = require('events').EventEmitter;
 
+var getBackLinks = function (previousSteps) {
+    return proxyquire('../../lib/middleware/back-links', {
+        '../util/helpers': {
+            getPreviousSteps: sinon.stub().returns(previousSteps)
+        }
+    });
+};
+
```

Comment:
if you require the helpers and stub the method then it will only be stubbed in the scope of the tests - when requiring the middleware to test, this in turn requires the unstubbed method from helpers. Proxyquire allows you to specify which sub deps to stub, unless I am missing something? @JoeChapman ?